server/asset/eth/eth.go

```diff
@@ -687,57 +678,18 @@ func (eth *ETHBackend) poll(ctx context.Context) {
 			be.sendBlockUpdate(u)
 		}
 	}
-	bhdr, err := eth.node.bestHeader(ctx)
-	if err != nil {
-		send(false, fmt.Errorf("error getting best block header from geth: %w", err))
-		return
-	}
-	if bhdr.Hash() == best.hash {
-		// Same hash, nothing to do.
-		return
-	}
-	update := func(reorg bool, fastBlocks bool) {
-		hash := bhdr.Hash()
-		height := bhdr.Number.Uint64()
-		str := fmt.Sprintf("Tip change from %s (%d) to %s (%d).",
-			best.hash, best.height, hash, height)
-		switch {
-		case reorg:
-			str += " Detected reorg."
-		case fastBlocks:
-			str += " Fast blocks."
-		}
-		eth.log.Debug(str)
-		best.hash = hash
-		best.height = height
-	}
-	if bhdr.ParentHash == best.hash {
-		// Sequential hash, report a block update.
-		update(false, false)
-		send(false, nil)
-		return
-	}
-	// Either a block was skipped or a reorg happened. We can only detect
-	// the reorg if our last best header's hash has changed. Otherwise,
-	// assume no reorg and report the new block change.
-	//
-	// headerByHeight will only return mainchain headers.
-	hdr, err := eth.node.headerByHeight(ctx, best.height)
+	bn, err := eth.node.blockNumber(ctx)
 	if err != nil {
-		send(false, fmt.Errorf("error getting block header from geth: %w", err))
+		send(fmt.Errorf("error getting best block header from geth: %w", err))
 		return
 	}
-	if hdr.Hash() == best.hash {
-		// Our recorded hash is still on main chain so there is no reorg
-		// that we know of. The chain has advanced more than one block.
-		update(false, true)
-		send(false, nil)
+	if bn == eth.bestHeight {
+		// Same height, nothing to do.
 		return
 	}
-	// The block for our recorded hash was forked off and the chain had a
-	// reorganization.
-	update(true, false)
-	send(true, nil)
+	eth.log.Debugf("Tip change from %d to %d.", eth.bestHeight, bn)
```

Comment:
Final note that tip change log no longer includes the block hashes, and I'm fine with that.
If we really need the reorg bool and the best block hashes back, we can keep polling using `eth_blockNumber`, and put back the `eth.node.headerByHeight` call if the height has changed.  At least the frequency will be more like 12 seconds instead of 1 sec.